src/ecdsa/keys.py

```diff
@@ -1133,3 +1133,28 @@ def sign_number(self, number, entropy=None, k=None):
         assert 1 <= _k < order
         sig = self.privkey.sign(number, _k)
         return sig.r, sig.s
+
+    def ecdh_get_shared_secret(self, remote_public_key):
+        """
+        Elliptic-curve Diffie-Hellman (ECDH). A key agreement protocol.
+        Allows two parties, each having an elliptic-curve public-private key 
+        pair, to establish a shared secret over an insecure channel
+                
+        :param remote_public_key: public key of remote party (VerifyingKey)
+        :return: shared secret
+        """""
+
+        if not self.curve.curve.contains_point(
+                remote_public_key.pubkey.point.x(),
+                remote_public_key.pubkey.point.y()):
+            raise ValueError("Invalid curve in public key.")
+
+        # shared secret = PUBKEYtheirs * PRIVATEKEYours
+        result = remote_public_key.pubkey.point * self.privkey.secret_multiplier
+        if result == ellipticcurve.INFINITY:
+            raise ValueError("Invalid shared secret.")
```

Comment:
a custom exception (both here and above) would be more robust, something like `InvalidKeyShareError`